sw-emulator/lib/periph/src/soc_reg.rs

```diff
@@ -441,6 +443,9 @@ struct SocRegistersImpl {
     /// Timer
     timer: Timer,
 
+    /// Log Directory
+    log_dir: PathBuf,
```

Comment:
Why does this need to be here? Can't the callback just capture this in the app?

Reply:
Maybe this is used somewhere else in the emulator library?

Reply:
Updated.